controller/src/beerocks/master/db/db.h

```diff
@@ -463,6 +464,24 @@ class db {
     bool get_node_channel_ext_above_secondary(std::string mac);
     uint16_t get_hostap_vht_center_frequency(std::string mac);
 
+    //
+    // certification
+    //
+    std::shared_ptr<uint8_t> &get_certification_tx_buffer() { return certification_tx_buffer; };
+    std::shared_ptr<uint8_t> &allocate_certification_tx_buffer()
+    {
+        return (certification_tx_buffer =
+                    std::shared_ptr<uint8_t>(new uint8_t[beerocks::message::MESSAGE_BUFFER_LENGTH],
```

Comment:
I don't understand why this buffer is needed, and I *defnitely* don't understand why it needs to be MESSAGE_BUFFER_LENGTH instead of 1 byte.

Reply:
We use the same buffer length of the controller tx_buffer, which is actually declared in beerocks_socket_thread. The reason for the size is because we copy the cmdu_tx buffer (tx_buffer) to the certification buffer when the certification flag is set.